lib/DownloadUtils.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Created by AsherS on 8/20/15.
+ */
+
+var fs = require('fs');
+var request = require('request');
+var Q = require('q');
+
+var DownloadUtils = (function(){
+
+    function DownloadUtils(){
```

Comment:
You have a function which when you call returns a function whose sole purpose is to host the "downloadFile" function - this is unnecessarily complex - why not just return an object with a downloadFile function (assuming you decide not to use q-http)?